engine/engine.go

```diff
@@ -233,24 +227,30 @@ func (e *EngineAPI) GetPayloadV3(payloadID engine.PayloadID) (*eth.ExecutionPayl
 	e.lock.RLock()
 	defer e.lock.RUnlock()
 
-	payload := e.payloadStore.Current()
-	if currentID := payload.ID(); payloadID != *currentID {
+	if e.currentPayloadAttributes == nil {
+		return nil, engine.InvalidParams.With(errors.New("engine payload is nil"))
```

Comment:
```suggestion
		return nil, engine.InvalidParams.With(errors.New("payload not found"))
```
nit: error message that may make more sense to the user, and indicates that we haven't receive the attributes from forkchoiceUpdated